packages/runtime/src/errorHandler.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  ClientComputationFunction,
+  Dependency,
+  ServerComputationFunction,
+  SSRComputationFunction,
+} from "./utils";
+
+export class InternalUseSSRComputationError extends Error {
+  result: any;
+  constructor(message: string, result) {
+    super(message);
+    this.name = 'InternalUseSSRComputationError';
+    this.result = result;
```

Comment:
It still needs to return a result when an error happens to make the hook return the same result on the server and client sides. (prevent causing conflicts while hydration).

Reply:
For example, the `Unsupported Async computation on server` error is only thrown on the server side, but the client side will return a normal `Promise`. So, I need to make the server-side returns a promise as well.